runtime/jailer.go

```diff
@@ -58,6 +58,10 @@ type jailer interface {
 	StubDrivesOptions() []stubDrivesOpt
 }
 
+type cgroupPather interface {
+	CgroupPath() string
+}
```

Comment:
Why not just add `CgroupPath` to the `jailer` interface? It seems a lot simpler to just put it there given both of our implementations have methods for it anyways.

Reply:
cgroup path doesn't feel like it belongs in a jailer interface and seems to be more of a implementation detail which is why I made separate. The `noop` jailer doesn't implement it, and if we decided to use Firecracker's jailer, it would also be absent.

Reply:
If so, shouldn't we remove CgroupPath from noopJailer?

Reply:
Thought I removed it, but I guess not o.O

Anyways, fixed!

Reply:
I'm confused why `CgroupPath` wouldn't belong in the jailer, it is a piece of information that the jailer configures as part of creating the jail and that we expose to users by sending it w/ the CreateVMResp and GetVMInfoResp (which makes it more than just an implementation detail). If we created a common jailing interface, then I would just imagine that Firecracker's jailer implementation would just return `""`, like the noop jailer, since it doesn't configure them. It seems like exactly what should be in a jailer interface.

Reply:
It's just that. Only the runc jailer sets the value. Why put it in a **common** interface, if it is only used for one of the jailer types?

Reply:
Because the other types are explicitly not setting it, which is just as relevant as it being set by the runc implementation. It's something that can be configured by the general concept of a "jailer" (and is by many), so it is confusing for it not to be part of a "jailer" interface.

Reply:
They aren't setting it because it doesn't have a notion of a cgroup path which I have hard time following to why it should be in the `jailer` interface.

Reply:
@samuelkarp @nmeyerhans - any preference on this?

Reply:
The lack of cgroups can be represented by either not implementing an interface (cgroupPather) or returning a special value (""). Using an interface may be more scalable if we have a lot of differences between jailers, where implementing all of required methods to just say "no" is hassle.

But, we haven't reached that stage, and we may not. So I'm fine with both ways :)

Reply:
If no one else has any feelings either way on this, then given this is a subjective issue I think the rule of thumb should be to go with the person who actually wrote the code :-)

The only other thing I'd add is that part of the reason this causes me deep confusion is that it seems to negate the existence of the jailer interface. If we want to have separate interfaces for each possible feature of a jailer, then why would we even have a jailer interface at all, especially given we have a no-op jailer implementation? I personally greatly prefer the simplicity and explicitness of just maintaining a single interface here (even if it's slightly more verbose), but if we want to split up features into separate interfaces then I think we should at least be consistent in doing so, which would suggest that we get rid of the no-op jailer and replace it with interface tests that have a default code path in case the method doesn't exist.

Reply:
@sipsma - I am thinking if there is a solution that could make both of us happy. So I was thinking perhaps adding something like `SetCreateVMResponse`. The only issue I have with this is you'd have to look at what the jailer is overwriting when setting any of the responses fields, but I think that may be fine. What are your thoughts on this approach?

Reply:
Honestly I like that option less than using `cgroupPather` because it ties the jailer directly into firecracker-containerd's api, whereas the eventual end goal in my mind is to have a generic jailer interface that's independent of firecracker-containerd and lives in the go sdk. But like I said before, these are stylistic concerns and they can be modified in the future pretty easily if we agree to, so I don't want all of this to block the PR if there are no feelings either way from anyone else right now. We can go with your original option or the CreateVMResponse-based option for now; when we refactor the jailer some more in the future would be a good time to discuss all this further.